packages/@uppy/react/src/DashboardModal.ts

```diff
@@ -0,0 +1,275 @@
+import { createElement as h, Component } from 'react'
+import PropTypes from 'prop-types'
+import DashboardPlugin from '@uppy/dashboard'
+import type { PluginTarget } from '@uppy/core/lib/UIPlugin'
+import type { Body, Meta } from '@uppy/utils/lib/UppyFile'
+import {
+  cssSize,
+  locale,
+  metaFields,
+  plugins,
+  uppy as uppyPropType,
+} from './propTypes.ts'
+import getHTMLProps from './getHTMLProps.ts'
+import nonHtmlPropsHaveChanged from './nonHtmlPropsHaveChanged.ts'
+import type { DashboardProps } from './Dashboard.ts'
+
+interface DashboardModalProps<M extends Meta, B extends Body>
+  extends DashboardProps<M, B> {
+  animateOpenClose?: boolean
+  disableLocalFiles?: boolean
+  autoOpenFileEditor?: boolean
+  browserBackButtonClose?: boolean
+  showRemoveButtonAfterComplete?: boolean
+  showSelectedFiles?: boolean
+  theme?: 'auto' | 'dark' | 'light'
+  waitForThumbnailsBeforeUpload?: boolean
+  fileManagerSelectionType?: 'files' | 'folders' | 'both'
+  hideRetryButton?: boolean
+  showLinkToFileUploadResult?: boolean
+  closeAfterFinish?: boolean
+  disabled?: boolean
+  target?: PluginTarget<M, B>
+  open?: boolean
+  onRequestClose: () => void
+  closeModalOnClickOutside?: boolean
+  disablePageScrollWhenModalOpen?: boolean
+  hideCancelButton?: boolean
+  hidePauseResumeButton?: boolean
```

Comment:
Ideally, we have three exported types in `@uppy/dashboard`, one for the model, one for inline, and one discriminated union based on the `inline` type which decides which types are present.

Then in all frameworks we import these different types and type the models with a single source of truth. Otherwise we have to write this list of types for every framework and they get out of sync.